e2etests/test/testUtils.js

```diff
@@ -67,13 +67,19 @@ function closeDeviceEventHubClients(deviceClient, eventHubClient, ehReceivers, d
       done(eventHubErr);
     }
   });
-  deviceClient.close(function (err) {
-    deviceErr = err || eventHubErr;
-    deviceClient = null;
-    if (deviceErr || !eventHubClient) {
-      done(deviceErr);
+  if (!deviceClient) {
+    if (!eventHubClient) {
+      done();
     }
```

Comment:
shouldn't this come before the promise.map call?
in any case, looks like we should split this function.